lib/urldata.h

```diff
@@ -1306,8 +1306,7 @@ struct UrlState {
     char *proxypasswd;
 #endif
   } aptr;
-  unsigned char httpwant; /* when non-zero, a specific HTTP version requested
-                             to be used in the library's request(s) */
+  unsigned char httpv_mask; /* bitmask of wanted CURL_HTTPV_* versions */
```

Comment:
Alternatively, we make each HTTP version a bit by itself:
~~~c
BIT(http_v1_0);
BIT(http_v1_1);
BIT(http_v2);
BIT(http_v3);
~~~

It might make the code nicer to read as it avoid a lot of bitwise ANDs. But instead adds some macro requirement if you want to check for multiple versions.